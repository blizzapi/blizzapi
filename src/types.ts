/** Common */

export type Uri = string;

/** Generic */

export enum ConstantKey {
  REGIONS = "REGIONS",
  LOCALES = "LOCALES",
  DEFAULT_LOCALES = "DEFAULT_LOCALES",
  REGION_API_HOSTS = "REGION_API_HOSTS",
  SC2_REALMS = "SC2_REALMS",
  OAUTH_AUTHORIZE_URIS = "OAUTH_AUTHORIZE_URIS",
  OAUTH_TOKEN_URIS = "OAUTH_TOKEN_URIS",
  OAUTH_CHECK_TOKEN_URIS = "OAUTH_CHECK_TOKEN_URIS",
}

/** Regions */

export enum RegionName {
  us = "us",
  eu = "eu",
  kr = "kr",
  tw = "tw",
  cn = "cn",
}

export enum RegionId {
  us = 1,
  eu = 2,
  kr = 3,
  tw = 3,
  cn = 5,
}

export enum RegionIdAsString {
  us = "1",
  eu = "2",
  kr = "3",
  tw = "3",
  cn = "5",
}

export type RegionIdKey = string;
export type RegionIdAsNumberOrString = RegionId | RegionIdAsString;
export type RegionIdArray = ReadonlyArray<RegionId>;
export type RegionNameArray = ReadonlyArray<RegionName>;

/** Locales */

export enum Locale {
  en_US = "en_US",
  es_MX = "es_MX",
  pt_BR = "pt_BR",
  en_GB = "en_GB",
  es_ES = "es_ES",
  fr_FR = "fr_FR",
  ru_RU = "ru_RU",
  de_DE = "de_DE",
  pt_PT = "pt_PT",
  it_IT = "it_IT",
  ko_KR = "ko_KR",
  zh_TW = "zh_TW",
  zh_CN = "zh_CN",
}

export type LocaleArray = ReadonlyArray<Locale>;

/** Default locales */

export enum DefaultLocaleIndex {
  us = 0,
  eu = 0,
  kr = 0,
  tw = 0,
  cn = 0,
}

export type DefaultLocaleIndexArray = ReadonlyArray<DefaultLocaleIndex>;

/** StarCraft II realms  */

export enum Sc2Realm {
  US = 1,
  LatAm = 2,
  Europe = 1,
  Russia = 2,
  Korea = 1,
  Taiwan = 2,
  China = 1,
}

export enum Sc2RealmAsString {
  US = "1",
  LatAm = "2",
  Europe = "1",
  Russia = "2",
  Korea = "1",
  Taiwan = "2",
  China = "1",
}

export type Sc2RealmArray = ReadonlyArray<Sc2Realm>;
export type Sc2RealmList = ReadonlyArray<Sc2Realm>;
export type Sc2RealmAsNumberOrString = Sc2Realm | Sc2RealmAsString;

/** API class types */

export type RegionIdOrName = RegionId | RegionName;
export type MaybeRegion = RegionIdOrName | undefined;
export type ClientId = string;
export type ClientSecret = string;
export type AccessToken = string | undefined;
export type Endpoint = string;

export enum HttpMethod {
  GET = "GET",
  POST = "POST",
}

export type ValidatorFunction = (endpoint: Endpoint) => boolean;

/* OAuth flow URIs */

export enum OAuthHost {
  us = "https://us.battle.net",
  eu = "https://eu.battle.net",
  kr = "https://apac.battle.net",
  tw = "https://apac.battle.net",
  cn = "https://www.battlenet.com.cn",
}

export enum RegionHost {
  us = "https://us.api.blizzard.com",
  eu = "https://eu.api.blizzard.com",
  kr = "https://kr.api.blizzard.com",
  tw = "https://tw.api.blizzard.com",
  cn = "https://gateway.battlenet.com.cn",
}

export enum OAuthEndpoint {
  authorize = "/oauth/authorize",
  token = "/oauth/token",
  checkToken = "/oauth/check_token?token=",
}

/** API headers */

export enum ApiHeaders {
  LastModified = "last-modified",
}

/** Errors */

export enum ErrorCode {
  NotAuthorized = 401,
}

export enum ErrorResponseMessage {
  InvalidToken = "invalid_token",
  AccessTokenInvalid = "access_token_invalid",
  AccessTokenExpired = "access_token_invalid",
}

export type InvalidAccessTokenError = {
  error: ErrorResponseMessage.AccessTokenInvalid;
};

export type ExpiredAccessTokenError = {
  error: ErrorResponseMessage.AccessTokenExpired;
};

export type ResponseError = InvalidAccessTokenError | ExpiredAccessTokenError;

/** Interfaces */

export interface AccessTokenOptions {
  validateAccessTokenOnEachQuery?: boolean;
  refreshExpiredAccessToken?: boolean;
  onAccessTokenExpired?: () => void;
  onAccessTokenRefresh?: (newAccessToken: string) => void;
}

export interface QueryOptions {
  timeout?: number;
  headers?: object | Headers;
  params?: object | URLSearchParams;
}

export interface BattleNetOptions {
  region: RegionIdOrName;
  clientId: ClientId;
  clientSecret: ClientSecret;
  accessToken?: AccessToken;
}

export interface InitOptions {
  region: string | number;
  clientId?: string;
  clientSecret?: string;
  accessToken?: string;
  validateAccessTokenOnEachQuery?: boolean;
  revalidateAccessTokenIfExpired?: boolean;
  onAccessTokenExpire?: () => void;
  onAccessTokenRevalidate?: () => void;
}

export interface BattleNetQueryOptions {
  region: RegionIdOrName;
  endpoint: string;
  clientId: string;
  clientSecret: string;
  accessToken: AccessToken;
  options: AccessTokenOptions & QueryOptions;
}

export type RegionIdProperties<Value> = {
  [key in RegionId]: Value;
};

export interface BlizzAPIOptions extends BattleNetOptions, AccessTokenOptions {}
