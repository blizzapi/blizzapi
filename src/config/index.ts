/**
 * Class exposing configuration constants
 *
 * Via:
 *
 * https://develop.battle.net/documentation/guides/using-oauth
 * https://develop.battle.net/documentation/guides/regionality-partitions-and-localization
 */

import * as localization from 'config/localization';
import * as api from 'config/api';
import oauth from 'config/oauth';
import { ConfigObject } from 'types';

const config = {
  REGIONS: localization.regions,
  LOCALES: localization.locales,
  DEFAULT_LOCALES: localization.defaultLocales,
  SC2_REALMS: localization.sc2Realms,
  REGION_API_HOSTS: api.regionHosts,
  OAUTH_AUTHORIZE_URIS: oauth.authorizeUris,
  OAUTH_TOKEN_URIS: oauth.tokenUris,
  OAUTH_CHECK_TOKEN_URIS: oauth.checkTokenUris,
} as ConfigObject;

export default config;
