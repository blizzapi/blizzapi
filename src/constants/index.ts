/**
 * Class exposing configuration constants
 *
 * Via:
 *
 * https://develop.battle.net/documentation/guides/using-oauth
 * https://develop.battle.net/documentation/guides/regionality-partitions-and-localization
 */

import {
  ConstantKey,
  DefaultLocaleIndex,
  LocaleArray,
  RegionIdProperties,
  RegionNameArray,
  Sc2RealmArray,
} from "../types";
import * as localization from "./localization";
import * as api from "./api";
import * as oauth from "./oauth";

export const constants = {
  [ConstantKey.REGIONS]:
    localization.regions as RegionIdProperties<RegionNameArray>,
  [ConstantKey.LOCALES]:
    localization.locales as RegionIdProperties<LocaleArray>,
  [ConstantKey.DEFAULT_LOCALES]:
    localization.defaultLocales as RegionIdProperties<DefaultLocaleIndex>,
  [ConstantKey.SC2_REALMS]:
    localization.sc2Realms as RegionIdProperties<Sc2RealmArray>,
  [ConstantKey.REGION_API_HOSTS]: api.regionHosts,
  [ConstantKey.OAUTH_AUTHORIZE_URIS]: oauth.authorizeUris,
  [ConstantKey.OAUTH_TOKEN_URIS]: oauth.tokenUris,
  [ConstantKey.OAUTH_CHECK_TOKEN_URIS]: oauth.checkTokenUris,
};
