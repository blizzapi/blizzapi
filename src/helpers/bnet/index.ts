// eslint-disable-next-line import/prefer-default-export
export { default as query } from './query';
