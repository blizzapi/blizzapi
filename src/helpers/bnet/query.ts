import { endpoint as validateEndpoint } from '../validators';
import { getAccessToken, validateAccessToken } from '../oauth';
import { getApiHostByRegion } from '../../utils/api';
import { fetchFromUri } from '../fetch';
import {
  RegionIdOrName,
  AccessToken,
  AccessTokenOptions,
  QueryOptions,
} from '../../types';

interface BattleNetQueryOptions {
  region: RegionIdOrName;
  endpoint: string;
  clientId: string;
  clientSecret: string;
  accessToken: AccessToken;
  options: AccessTokenOptions & QueryOptions;
}

const queryWithAccessToken = (queryOptions: BattleNetQueryOptions, accessToken: AccessToken) => {
  const {
    region,
    endpoint,
    options,
  } = queryOptions;
  const {
    headers,
    params,
    timeout,
  } = options;
  const validEndpoint = validateEndpoint(endpoint);
  if (!validEndpoint) throw new RangeError(`${endpoint} is not a valid endpoint.`);

  const apiHost = getApiHostByRegion(region);
  const requestUri = `${apiHost}${endpoint}`;
  const authHeaders = {
    Authorization: `Bearer ${accessToken}`,
  };

  const fetchHeaders = {
    ...headers,
    ...authHeaders,
  };

  return fetchFromUri({
    uri: requestUri,
    method: 'GET',
    headers: fetchHeaders,
    ...params && { params },
    ...timeout && { timeout },
  });
};

export const query = async (queryOptions: BattleNetQueryOptions) => {
  const { region, accessToken } = queryOptions;
  const {
    validateAccessTokenOnEachQuery,
    refreshExpiredAccessToken,
    onAccessTokenExpired,
    onAccessTokenRefresh,
  } = queryOptions.options;

  if (validateAccessTokenOnEachQuery) {
    const invalidAccessToken = !(await validateAccessToken(region, accessToken));
    if (invalidAccessToken) {
      return {
        error: 'access_token_invalid',
      };
    }
  }

  try {
    return await queryWithAccessToken(queryOptions, accessToken);
  } catch (error: any) {
    if (error.response && error.response.status === 401) {
      onAccessTokenExpired?.();
      if (refreshExpiredAccessToken) {
        const newAccessToken = await getAccessToken(queryOptions);
        onAccessTokenRefresh?.(newAccessToken);
        return queryWithAccessToken(queryOptions, newAccessToken);
      }
      return Promise.resolve({
        error: 'access_token_invalid',
      });
    }
    throw error;
  }
};
