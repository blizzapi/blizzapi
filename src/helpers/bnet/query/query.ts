import {
  RegionIdOrName,
  AccessToken,
  AccessTokenOptions,
} from '../../../../types';
import { endpoint as validateEndpoint } from '../validators';
import { getAccessToken, validateAccessToken }  from '../oauth';
import { getApiHostByRegion } from '../../../utils/api';
import fetchFromUri from '../../fetch/fetchFromUri/fetchFromUri';

interface QueryOptions {
  region: RegionIdOrName,
  endpoint: string,
  clientId: string,
  clientSecret: string,
  accessToken: AccessToken,
  options: AccessTokenOptions,
}

const queryWithAccessToken = (queryOptions: QueryOptions, accessToken: AccessToken) => {
  const { region, endpoint } = queryOptions;
  const validEndpoint = validateEndpoint(endpoint);
  if (!validEndpoint) throw new RangeError(`${endpoint} is not a valid endpoint.`);

  const apiHost = getApiHostByRegion(region);
  const requestUri = `${apiHost}${endpoint}`;
  const headers = {
    Authorization: `Bearer ${accessToken}`,
  };

  return fetchFromUri({
    headers,
    uri: requestUri,
    method: 'GET',
  });
}

export default async (queryOptions: QueryOptions) => {
  const { region, accessToken } = queryOptions;
  const {
    validateAccessTokenOnEachQuery,
    refreshExpiredAccessToken,
    onAccessTokenExpired,
    onAccessTokenRefresh,
  } = queryOptions.options;

  if (validateAccessTokenOnEachQuery) {
    const invalidAccessToken = !(await validateAccessToken(region, accessToken));
    if (invalidAccessToken) {
      return {
        error: 'access_token_invalid',
      }
    }
  }

  try {
    const data = await queryWithAccessToken(queryOptions, accessToken);
    return data;
  } catch (error) {
    if (error.response && error.response.status === 401) {
      onAccessTokenExpired && onAccessTokenExpired();
      if (refreshExpiredAccessToken) {
        const newAccessToken = await getAccessToken(queryOptions);
        onAccessTokenRefresh && onAccessTokenRefresh(newAccessToken);
        return queryWithAccessToken(queryOptions, newAccessToken);
      }
      return Promise.resolve({
        error: 'access_token_invalid',
      });
    }
    throw error;
  }
};
