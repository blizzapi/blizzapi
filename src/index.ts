import BlizzAPI from './lib/classes/BlizzAPI';
export = BlizzAPI;