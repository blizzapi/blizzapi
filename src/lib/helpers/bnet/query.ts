import { RegionIdOrName, AccessToken } from '../../../../@types';
import { endpoint as validateEndpoint } from '../validators';
import { getApiHostByRegion } from '../../utils/api';
import { fetchFromUri } from '../fetch';
// import { BlizzAPIOptions } from '../../classes/BlizzAPI';


export default (
  region: RegionIdOrName,
  endpoint: string,
  accessToken: AccessToken,
  // options?: BlizzAPIOptions,
) => {
  const validEndpoint = validateEndpoint(endpoint);
  if (!validEndpoint) throw new RangeError(`${endpoint} is not a valid endpoint.`);

  const apiHost = getApiHostByRegion(region);
  const requestUri = `${apiHost}${endpoint}`;
  const headers = {
    Authorization: `Bearer ${accessToken}`,
  }

  return fetchFromUri(requestUri, 'GET', headers);
};
