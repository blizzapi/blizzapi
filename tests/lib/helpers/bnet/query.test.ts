import { query } from "../../../../src/helpers/bnet/query";
import { RegionName } from "../../../../src/types";

jest.mock("../../../../src/helpers/oauth");
// eslint-disable-next-line @typescript-eslint/no-var-requires
const oauth = require("../../../../src/helpers/oauth");

oauth.validateAccessToken = jest
  .fn()
  // eslint-disable-next-line no-empty-pattern
  .mockImplementation(({}, accessToken: string) => {
    if (accessToken === "invalid_access_token") return Promise.resolve(false);
    return Promise.resolve(true);
  });

oauth.getAccessToken = jest
  .fn()
  .mockImplementation(() => Promise.resolve("new_refreshed_access_token"));

describe("query()", () => {
  it("should be defined", () => {
    expect(query).toBeDefined();
  });

  it("should be function", () => {
    expect(query).toBeTruthy();
    expect(typeof query).toBe("function");
  });

  it("returns correct response for GET request to a valid uri", async () => {
    expect.assertions(1);
    const response = await query({
      region: "us" as RegionName,
      endpoint: "/sc2/sample/endpoint",
      clientId: "valid_client_id",
      clientSecret: "valid_client_secret",
      accessToken: "valid_access_token",
      options: {
        validateAccessTokenOnEachQuery: false,
        refreshExpiredAccessToken: false,
        onAccessTokenExpired: undefined,
        onAccessTokenRefresh: undefined,
      },
    });
    expect(response).toMatchSnapshot();
  });

  it("returns correct response for GET request with params to a valid uri", async () => {
    expect.assertions(1);
    const response = await query({
      region: "us" as RegionName,
      endpoint: "/sc2/sample/endpoint",
      clientId: "valid_client_id",
      clientSecret: "valid_client_secret",
      accessToken: "valid_access_token",
      options: {
        validateAccessTokenOnEachQuery: false,
        refreshExpiredAccessToken: false,
        onAccessTokenExpired: undefined,
        onAccessTokenRefresh: undefined,
        params: {
          data: "test params",
        },
      },
    });
    expect(response).toMatchSnapshot();
  });

  it("rejects and throws RangeError for invalid endpoint", async () => {
    expect.assertions(1);
    expect(
      query({
        region: "us" as RegionName,
        endpoint: "invalidEndpoint",
        clientId: "valid_client_id",
        clientSecret: "valid_client_secret",
        accessToken: "valid_access_token",
        options: {
          validateAccessTokenOnEachQuery: false,
          refreshExpiredAccessToken: false,
          onAccessTokenExpired: undefined,
          onAccessTokenRefresh: undefined,
        },
      })
    ).rejects.toThrow(RangeError);
  });

  it("validates access token if validateAccessTokenOnEachQuery is set to true", async () => {
    expect.assertions(1);
    const response = await query({
      region: "us" as RegionName,
      endpoint: "/valid/endpoint",
      clientId: "valid_client_id",
      clientSecret: "valid_client_secret",
      accessToken: "valid_access_token",
      options: {
        validateAccessTokenOnEachQuery: true,
        refreshExpiredAccessToken: false,
        onAccessTokenExpired: undefined,
        onAccessTokenRefresh: undefined,
      },
    });

    expect(response).toMatchSnapshot();
  });

  it("returns error if validateAccessTokenOnEachQuery is set to true and access token is invalid", async () => {
    expect.assertions(1);
    const response = await query({
      region: "us" as RegionName,
      endpoint: "/valid/endpoint",
      clientId: "valid_client_id",
      clientSecret: "valid_client_secret",
      accessToken: "invalid_access_token",
      options: {
        validateAccessTokenOnEachQuery: true,
        refreshExpiredAccessToken: false,
        onAccessTokenExpired: undefined,
        onAccessTokenRefresh: undefined,
      },
    });
    expect(response).toMatchSnapshot();
  });

  it("returns error if validateAccessTokenOnEachQuery is set to false and access token is invalid", async () => {
    expect.assertions(1);
    const response = await query({
      region: "us" as RegionName,
      endpoint: "/valid/endpoint",
      clientId: "valid_client_id",
      clientSecret: "valid_client_secret",
      accessToken: "invalid_access_token",
      options: {
        validateAccessTokenOnEachQuery: false,
        refreshExpiredAccessToken: false,
        onAccessTokenExpired: undefined,
        onAccessTokenRefresh: undefined,
      },
    });
    expect(response).toMatchSnapshot();
  });

  it("calls onAccessTokenExpired() if provided and access token is invalid", async () => {
    expect.assertions(2);
    const onAccessTokenExpired = jest.fn();
    const response = await query({
      region: "us" as RegionName,
      endpoint: "/valid/endpoint",
      clientId: "valid_client_id",
      clientSecret: "valid_client_secret",
      accessToken: "invalid_access_token",
      options: {
        onAccessTokenExpired,
        validateAccessTokenOnEachQuery: false,
        refreshExpiredAccessToken: false,
        onAccessTokenRefresh: undefined,
      },
    });
    expect(response).toMatchSnapshot();
    expect(onAccessTokenExpired).toHaveBeenCalledTimes(1);
  });

  it("refreshes access token if refreshExpiredAccessToken is set to true", async () => {
    expect.assertions(1);
    const response = await query({
      region: "us",
      endpoint: "/valid/endpoint",
      clientId: "valid_client_id",
      clientSecret: "valid_client_secret",
      accessToken: "invalid_access_token",
      options: {
        validateAccessTokenOnEachQuery: false,
        refreshExpiredAccessToken: true,
        onAccessTokenExpired: undefined,
        onAccessTokenRefresh: undefined,
      },
    });
    expect(response).toMatchSnapshot();
  });

  it("calls onAccessTokenRefresh() if provided and refreshExpiredAccessToken is set to true", async () => {
    expect.assertions(2);
    const onAccessTokenRefresh = jest.fn();

    const response = await query({
      region: "us" as RegionName,
      endpoint: "/valid/endpoint",
      clientId: "valid_client_id",
      clientSecret: "valid_client_secret",
      accessToken: "invalid_access_token",
      options: {
        onAccessTokenRefresh,
        validateAccessTokenOnEachQuery: false,
        refreshExpiredAccessToken: true,
        onAccessTokenExpired: undefined,
      },
    });
    expect(response).toMatchSnapshot();
    expect(onAccessTokenRefresh).toHaveBeenCalledTimes(1);
  });

  it("respects timeout option", async () => {
    expect.assertions(1);

    const response = await query({
      region: "us" as RegionName,
      endpoint: "/valid/endpoint",
      clientId: "valid_client_id",
      clientSecret: "valid_client_secret",
      accessToken: "valid_access_token",
      options: {
        timeout: 123456,
      },
    });
    expect(response).toMatchSnapshot();
  });
});
