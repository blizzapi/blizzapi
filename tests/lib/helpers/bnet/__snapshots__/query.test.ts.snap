// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query() calls onAccessTokenExpired() if provided and access token is invalid 1`] = `
{
  "error": "access_token_invalid",
}
`;

exports[`query() calls onAccessTokenRefresh() if provided and refreshExpiredAccessToken is set to true 1`] = `
{
  "access_token": "new_refreshed_access_token",
  "headers": {
    "Authorization": "Bearer new_refreshed_access_token",
  },
  "params": {
    "data": "default sample params",
  },
  "timeout": 10000,
  "title": "sample mocked data for GET request to https://us.api.blizzard.com/valid/endpoint",
}
`;

exports[`query() refreshes access token if refreshExpiredAccessToken is set to true 1`] = `
{
  "access_token": "new_refreshed_access_token",
  "headers": {
    "Authorization": "Bearer new_refreshed_access_token",
  },
  "params": {
    "data": "default sample params",
  },
  "timeout": 10000,
  "title": "sample mocked data for GET request to https://us.api.blizzard.com/valid/endpoint",
}
`;

exports[`query() respects timeout option 1`] = `
{
  "access_token": "valid_access_token",
  "headers": {
    "Authorization": "Bearer valid_access_token",
  },
  "params": {
    "data": "default sample params",
  },
  "timeout": 123456,
  "title": "sample mocked data for GET request to https://us.api.blizzard.com/valid/endpoint",
}
`;

exports[`query() returns correct response for GET request to a valid uri 1`] = `
{
  "access_token": "valid_access_token",
  "headers": {
    "Authorization": "Bearer valid_access_token",
  },
  "params": {
    "data": "default sample params",
  },
  "timeout": 10000,
  "title": "sample mocked data for GET request to https://us.api.blizzard.com/sc2/sample/endpoint",
}
`;

exports[`query() returns correct response for GET request with params to a valid uri 1`] = `
{
  "access_token": "valid_access_token",
  "headers": {
    "Authorization": "Bearer valid_access_token",
  },
  "params": {
    "data": "test params",
  },
  "timeout": 10000,
  "title": "sample mocked data for GET request to https://us.api.blizzard.com/sc2/sample/endpoint",
}
`;

exports[`query() returns error if validateAccessTokenOnEachQuery is set to false and access token is invalid 1`] = `
{
  "error": "access_token_invalid",
}
`;

exports[`query() returns error if validateAccessTokenOnEachQuery is set to true and access token is invalid 1`] = `
{
  "error": "access_token_invalid",
}
`;

exports[`query() validates access token if validateAccessTokenOnEachQuery is set to true 1`] = `
{
  "access_token": "valid_access_token",
  "headers": {
    "Authorization": "Bearer valid_access_token",
  },
  "params": {
    "data": "default sample params",
  },
  "timeout": 10000,
  "title": "sample mocked data for GET request to https://us.api.blizzard.com/valid/endpoint",
}
`;
